app/models/account.rb

```diff
@@ -56,16 +59,16 @@ def self.cache_or_create_by!(user)
   def self.create_if_needed!(user)
     attrs = account_attrs_from_user(user)
 
-    accts = where(idme_uuid: attrs[:idme_uuid])
-            .where.not(idme_uuid: nil)
-            .or(
-              where(sec_id: attrs[:sec_id])
-              .where.not(sec_id: nil)
-            )
+    accts = idme_uuid_match(attrs[:idme_uuid]).or(sec_id_match(attrs[:sec_id]))
 
     if accts.length > 1
       data = accts.map(&:attributes)
       log_message_to_sentry('multiple Account records with matching ids', 'warning', data)
+      # sort the accounts so the ones with matching idme_uuid values come first
+      # in the array, this will provide the users with a more consistent
+      # experience in case they have multiple credentials to login with
+      # https://github.com/department-of-veterans-affairs/va.gov-team/issues/6702
+      accts = accts.sort_by { |a| a.idme_uuid == attrs[:idme_uuid] ? 0 : 1 }
     end
```

Comment:
@ericbuckley since you've already got the explanation here (which is great -- its not obvious whats going on w/o some context), maybe you could turn this into a (private?) method that describes the function -- 
```
accts = sort_with_idme_uuid_priority(accts)

# https://github.com/department-of-veterans-affairs/va.gov-team/issues/6702
def  sort_with_idme_uuid_priority(accts)
   if accts.length > 1
      data = accts.map(&:attributes)
      log_message_to_sentry('multiple Account records with matching ids', 'warning', data)
      accts = accts.sort_by { |a| a.idme_uuid == attrs[:idme_uuid] ? 0 : 1 }
end
``` 

or something similar. Would fix the complexity issue in codeclimate and make it easier to read